lib/radiator/directory/editor.ex

```diff
@@ -519,21 +519,45 @@ defmodule Radiator.Directory.Editor do
   end
 
   def update_audio(actor = %Auth.User{}, audio = %Audio{}, attrs) do
-    if has_permission(actor, audio, :edit) do
+    if can_access_audio(actor, audio, :edit) do
       Editor.Manager.update_audio(audio, attrs)
     else
       @not_authorized_match
     end
   end
 
   def delete_audio(actor = %Auth.User{}, audio = %Audio{}) do
-    if has_permission(actor, audio, :own) do
+    if can_access_audio(actor, audio, :own) do
       Editor.Manager.delete_audio(audio)
     else
       @not_authorized_match
     end
   end
 
+  # thought: we need this logic somewhere, shouldn't it be in the
+  # permission context as the general logic is already there?
+  # I feel like it makes more sense bringing objects without permissions,
+  # like Audio, into the permissions module, than handling permission
+  # logic here.
+  # But that would mean supporting has_permission/3 on subjects
+  # that do not actually have their own permissions.
+  defp can_access_audio(actor, audio, permission) do
```

Comment:
One the other hand, maybe it's fine keeping this locally as long as it's only used from withing this Editor module? I keep going back and forth on this one 🤔

Reply:
I like it separate as long as we don't want to have it abstracted. IMHO it is good that in permissions only the entities with actuall permissions are handeled. Any derived permissions living farther out sounds good to me in terms nobody gets confused if audio permissions are separate or not. And after all, @timpritlove isn't satisfied with the current permission model anyway, so this will probably end up being a bigger and different task anyways.